storagemarket/impl/provider_environments.go

```diff
@@ -32,6 +33,27 @@ type providerDealEnvironment struct {
 	p *Provider
 }
 
+func (p *providerDealEnvironment) RegisterShard(ctx context.Context, pieceCid cid.Cid, carPath string, eagerInit bool) error {
+	return shared.RegisterShardSync(ctx, p.p.dagStore, pieceCid, carPath, eagerInit)
+}
+
+func (p *providerDealEnvironment) CARv2Reader(carV2FilePath string) (*carv2.Reader, error) {
+	return carv2.OpenReader(carV2FilePath)
+}
+
+func (p *providerDealEnvironment) FinalizeReadWriteBlockstore(proposalCid cid.Cid, carPath string, rootCid cid.Cid) error {
+	bs, err := p.p.readWriteBlockStores.GetOrCreate(proposalCid.String(), carPath, rootCid)
```

Comment:
Shouldn't this be a Get?

Reply:
If so, the signature of this method can take only the proposalCid.

Reply:
Alright, so this can't be a Get because it's possible that we restarted, and we are not tracking the store any longer. Ideally we would track stores for all pending transfers.